### PR TITLE
[AP-3834] Add github action workflow for test and lint

### PR DESCRIPTION
**Pull request by jsugarman**
## What

Add a github action workflow for CI and CD for test and lint (only)

[Link to story](https://dsdmoj.atlassian.net/browse/AP-3834)

Implement continuous integration

## Checklist

Before you ask people to review this PR:

- Tests and linters should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Review by colinbruce — COMMENTED** [review]
Looks okay so far...
The commit says push to ECR but I can't see that... 
Will deploy be following soon?

**Comment by jsugarman** [comment]
> Looks okay so far... The commit says push to ECR but I can't see that... Will deploy be following soon?

we have decided to break them up inline with tickets and discussions around secrets management

File: .github/workflows/test-build-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        run: yarn install --frozen-lockfile
+
+      - name: Precompile assets
+        run: bin/rails assets:precompile
+
+      - name: Setup test database
+        run: bin/rails db:prepare
+
+      - name: Run linters
+        run: bundle exec rubocop
```
Comment (colinbruce) [inline .github/workflows/test-build-deploy.yml:65]: Would running the `rake` task hit _all_ linters and make this easier to read/harder to parse?
e.g. adding erblint to the rake task would run that as well as rubocop and rspec 🤷‍♂️

Comment (jsugarman) [reply .github/workflows/test-build-deploy.yml:65]: yep, can do that

Comment (jsugarman) [reply .github/workflows/test-build-deploy.yml:65]: i take it back rake runs everything including tests. will added erblint to the linters though. erblint was added after i wrote this.

**Review by kmahern — APPROVED** [review]
lgtm 👍

**Pull request merged**